cmd/traffic/cmd/manager/service.go

```diff
@@ -545,3 +546,27 @@ func (m *Manager) WatchLookupHost(session *rpc.SessionInfo, stream rpc.Manager_W
 func (m *Manager) expire() {
 	m.state.ExpireSessions(m.clock.Now().Add(-15 * time.Second))
 }
+
+// reapDomain informs SystemA that an intercept with a domain has been garbage collected
+func (m *Manager) reapDomain(ctx context.Context, sa systema.SystemACRUDClient, interceptUpdate watchable.InterceptMapUpdate) error {
+	dlog.Debugf(ctx, "systema: removing domain: %q", interceptUpdate.Value.PreviewDomain)
+	_, err := sa.RemoveDomain(ctx, &systema.RemoveDomainRequest{
+		Domain: interceptUpdate.Value.PreviewDomain,
+	})
+	return err
+}
+
+// reapIntercept informs SystemA that an intercept has been garbage collected
+func (m *Manager) reapIntercept(ctx context.Context, sa systema.SystemACRUDClient, interceptUpdate watchable.InterceptMapUpdate) error {
```

Comment:
Yeah although now that I'm looking at it again, I should probably check in this function that `interceptUpdate.Delete` is true... I know I do it before the function is called, but we don't want to call this function unless the intercept was indeed deleted, so I think I should code that into the functions in case they ever get used elsewhere.